connectors/sources/opentext_documentum.py

```diff
@@ -0,0 +1,491 @@
+            raise InternalServerError
+        else:
+            raise
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
+    async def api_call(self, url_name=None, **url_kwargs):
+        """Make a GET call for the Opentext Documentum API using the passed url_name with retry for the failed API calls.
+
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+            url_kwargs (dict): Url kwargs to format the query.
+
+        Raises:
+            exception: An instance of an exception class.
+
+        Yields:
+            response: Return api response.
+        """
+        url = url_kwargs.get("url") or parse.urljoin(
+            self.host_url, URLS[url_name].format(**url_kwargs)  # pyright: ignore
+        )
+        self._logger.debug(f"Making a GET call for url: {url}")
+        while True:
+            try:
+                async with self._get_session.get(
+                    url=url,
+                    ssl=self.ssl_ctx,
+                ) as response:
+                    yield response
+                    break
+            except ServerConnectionError:
+                self._logger.exception(f"Getting ServerConnectionError for url: {url}")
+                await self.close_session()
+                raise
+            except ClientResponseError as exception:
+                await self._handle_client_errors(url=url, exception=exception)
+
+    async def paginated_api_call(self, url_name, **kwargs):
+        """Make a paginated API call for Opentext Documentum objects using the passed url_name with retry for the failed API calls.
+
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+
+        Yields:
+            response: Return api response.
+        """
+        page = 0
+        self._logger.info(
+            f"Started pagination for API url: {parse.urljoin(self.host_url, URLS[url_name].format(page=page, items_per_page=ITEMS_PER_PAGE, **kwargs))} with {ITEMS_PER_PAGE} items per page"
+        )
+
+        while True:
+            try:
+                async for response in self.api_call(
+                    url_name=url_name,
+                    page=page,
+                    items_per_page=ITEMS_PER_PAGE,
+                    repository_name=kwargs.get("repository_name"),
+                    folder_id=kwargs.get("folder_id"),
+                ):
+                    response_json = await response.json()
+                    if not response_json.get("entries"):
+                        return
+                    yield response_json
+                    page += 1
+            except Exception as exception:
+                self._logger.warning(
+                    f"Skipping data for type: {url_name}, page: {page}. Error: {exception}."
+                )
+                break
```

Comment:
we used to skips the failed pages instead of raising the exception for that page and continues the execution for the next page. 
However, we can also re-raise an exception for the failed page.